python/ray/air/execution/_internal/actor_manager.py

```diff
@@ -0,0 +1,773 @@
+            uuid.uuid4().int, on_start=on_start, on_stop=on_stop, on_error=on_error
+        )
+
+        self._pending_actors_to_attrs[tracked_actor] = cls, kwargs, resource_request
+        self._resource_request_to_pending_actors[resource_request].append(tracked_actor)
+
+        self._resource_manager.request_resources(resource_request=resource_request)
+
+        return tracked_actor
+
+    def remove_actor(
+        self,
+        tracked_actor: TrackedActor,
+        kill: bool = False,
+    ) -> None:
+        """Remove a tracked actor.
+
+        If the actor has already been started, this will stop the actor. This will
+        trigger the :meth:`TrackedActor.on_stop
+        <ray.air.execution._internal.tracked_actor.TrackedActor.on_stop>`
+        callback once the actor stopped.
+
+        If the actor has only been requested, but not started, yet, this will cancel
+        the actor request. This will not trigger any callback.
+
+
+        If ``kill=True``, this will use ``ray.kill()`` to forcefully terminate the
+        actor. Otherwise, graceful actor deconstruction will be scheduled after
+        all currently tracked futures are resolved.
+
+        Args:
+            tracked_actor: Tracked actor to be removed.
+            kill: If set, will forcefully terminate the actor instead of gracefully
+                scheduling termination.
+        """
+        if tracked_actor in self._live_actors_to_ray_actors_resources:
+            # Ray actor is running.
+
+            if not kill:
+                # Schedule __ray_terminate__ future
+                ray_actor, _ = self._live_actors_to_ray_actors_resources[tracked_actor]
+
+                def on_actor_stop(*args, **kwargs):
+                    self._actor_stop_resolved(tracked_actor=tracked_actor)
+
+                stop_future = ray_actor.__ray_terminate__.remote()
+                self._actor_state_events.track_future(
+                    future=stop_future,
+                    on_result=on_actor_stop,
+                    on_error=on_actor_stop,
+                )
+                self._tracked_actors_to_state_futures[tracked_actor].add(stop_future)
+
+            else:
+                # kill = True
+                self._live_actors_to_kill.add(tracked_actor)
+
+        elif tracked_actor in self._pending_actors_to_attrs:
+            # Actor is pending, stop
+            _, _, resource_request = self._pending_actors_to_attrs.pop(tracked_actor)
+            self._resource_manager.cancel_resource_request(
+                resource_request=resource_request
+            )
+        else:
+            raise ValueError(f"Unknown tracked actor: {tracked_actor}")
+
+    def is_actor_started(self, tracked_actor: TrackedActor) -> bool:
+        """Returns True if the actor has been started.
+
+        Args:
+            tracked_actor: Tracked actor object.
+        """
+        return tracked_actor in self._live_actors_to_ray_actors_resources
```

Comment:
imagine we can just say ``return tracked_actor.started``. just a random idea.

Reply:
I thought about this! The reason why I'd like to keep it in the manager for now is that we can actually ship the tracked actor around. If we attached state to the tracked actor, we can't because we have to worry about consistency.

Let's see if we ever need to access state without having access to the manager directly - we can always add this property later.